v3/src/components/component-header.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { useState } from "react"
+import { Box, CloseButton, Flex } from "@chakra-ui/react"
+import { EditableComponentTitle } from "./editable-component-title"
+import t from "../utilities/translation/translate"
+import MinimizeIcon from "../assets/icons/icon-minimize.svg"
+import TableIcon from "../assets/icons/icon-table.svg"
+import CardIcon from "../assets/icons/icon-case-card.svg"
+
+import "./component-header.scss"
+
+interface IProps {
+  tileType: string
+  datasetName?: string
+}
+
+export const ComponentHeader = ({tileType, datasetName}: IProps) => {
+  const [componentTitle, setComponentTitle] = useState(datasetName || "New Dataset")
+  const [showSwitchMessage, setShowSwitchMessage] = useState(false)
+  const [showCaseCard, setShowCaseCard] = useState(false)
+
+  const handleTitleChange = (title?: string) => {
+    title && setComponentTitle(title)
+  }
+
+  const handleShowCardTableToggleMessage = () => {
+    setShowSwitchMessage(true)
+  }
+
+  const handleToggleCardTable = (e:React.MouseEvent) => {
+    e.stopPropagation()
+    setShowSwitchMessage(false)
+    setShowCaseCard(!showCaseCard)
+  }
+
+  const cardTableToggleString = showCaseCard
+                                  ? t("DG.DocumentController.toggleToCaseTable")
+                                  : t("DG.DocumentController.toggleToCaseCard")
+
+  return (
+    <Flex className="component-header">
+      {tileType === "codap-case-table" &&
```

Comment:
I'm uncomfortable with embedding tile specific knowledge in this general-purpose component. Is there any way around that that you can think of?

Reply:
I agree with Bill on this one. The common-code title bar should not have tile-specific code in it. Perhaps the client could pass in contents, e.g. as `children`. Take a look at how this is handled in the v2 code.